orderbook/src/fee.rs

```diff
@@ -0,0 +1,178 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use chrono::{DateTime, FixedOffset, Utc};
+use primitive_types::{H160, U256};
+use std::sync::Mutex;
+
+use crate::price_estimate::PriceEstimating;
+use gas_estimation::GasPriceEstimating;
+
+type Measurement = (U256, DateTime<Utc>);
+
+pub struct MinFeeCalculator {
+    price_estimator: Box<dyn PriceEstimating>,
+    gas_estimator: Box<dyn GasPriceEstimating>,
+    native_token: H160,
+    // TODO persist past measurements to shared storage
+    measurements: Mutex<HashMap<H160, Vec<Measurement>>>,
+    now: Box<dyn Fn() -> DateTime<Utc> + Send + Sync>,
+}
+
+const GAS_PER_ORDER: f64 = 100_000.0;
```

Comment:
Is this where we hard code the per trade gas cost average as benchmarked in the smart contract? Perhaps we can say a word about this here or include a link to the benchmarks?

Reply:
This was a rebase error, it's actually part of #292